modules/apps/transfer/types/packet_test.go

```diff
@@ -43,3 +44,213 @@ func TestFungibleTokenPacketDataValidateBasic(t *testing.T) {
 		}
 	}
 }
+
+func (suite *TypesTestSuite) TestGetSourceCallbackAddress() {
+	testCases := []struct {
+		name       string
+		packetData types.FungibleTokenPacketData
+		expPass    bool
+	}{
+		{
+			"memo is empty",
+			types.FungibleTokenPacketData{
+				Denom:    denom,
+				Amount:   amount,
+				Sender:   addr1,
+				Receiver: addr2,
+				Memo:     "",
+			},
+			false,
+		},
+		{
+			"memo is not json string",
+			types.FungibleTokenPacketData{
+				Denom:    denom,
+				Amount:   amount,
+				Sender:   addr1,
+				Receiver: addr2,
+				Memo:     "memo",
+			},
+			false,
+		},
+		{
+			"memo is does not have callbacks in json struct",
+			types.FungibleTokenPacketData{
+				Denom:    denom,
+				Amount:   amount,
+				Sender:   addr1,
+				Receiver: addr2,
+				Memo:     "{\"Key\": 10}",
+			},
+			false,
+		},
+		{
+			"memo has callbacks in json struct but does not have src_callback_address key",
+			types.FungibleTokenPacketData{
+				Denom:    denom,
+				Amount:   amount,
+				Sender:   addr1,
+				Receiver: addr2,
+				Memo:     "{\"callbacks\": {\"Key\": 10}}",
+			},
+			false,
+		},
+		{
+			"memo has callbacks in json struct but does not have string value for src_callback_address key",
+			types.FungibleTokenPacketData{
+				Denom:    denom,
+				Amount:   amount,
+				Sender:   addr1,
+				Receiver: addr2,
+				Memo:     "{\"callbacks\": {\"src_callback_address\": 10}}",
```

Comment:
don't think this is a big enough deal to change it, but I find using backticks can be more readable than requiring to escape quotes 

e.g.

```go
Memo: `{"callbacks": {"src_callback_address": 10}}`,
```

Reply:
Oh that's a lot cleaner. I will do this change when I get a chance (might not be until after this merge or monday)